components/brave_sync/crypto/crypto.cc

```diff
@@ -158,6 +159,15 @@ bool PassphraseToBytes32(const std::string& passphrase,
 }
 
 bool IsPassphraseValid(const std::string& passphrase) {
+  std::vector<std::string> words = base::SplitString(
+      passphrase, " ", base::WhitespaceHandling::TRIM_WHITESPACE,
+      base::SplitResult::SPLIT_WANT_NONEMPTY);
+  size_t num_words = words.size();
+  static constexpr size_t kPureWordsCount = 24u;
+  if (num_words != kPureWordsCount) {
+    return false;
+  }
```

Comment:
We should put a comment saying this check is dedicated for old client to reject sync code from new client which has time limited code (25 words)

Reply:
thanks, fixed at https://github.com/brave/brave-core/compare/d45af0114142144b9437d2980dc27bb52d734b7c..3bf3aa99a87396d1891508915517ffcb52442e08